plugins/org.jkiss.dbeaver.ext.databricks/src/org/jkiss/dbeaver/ext/databricks/DatabricksDataSource.java

```diff
@@ -74,4 +77,25 @@ protected Properties getAllConnectionProperties(
         connectionInfo.setProperty(DatabricksConstants.USER_AGENT_ENTRY, userAgent);
         return super.getAllConnectionProperties(monitor, context, purpose, connectionInfo);
     }
+
+    @Override
+    protected void initializeContextState(@NotNull DBRProgressMonitor monitor, @NotNull JDBCExecutionContext context,
+                                          @Nullable JDBCExecutionContext initFrom) throws DBException {
+        DatabricksExecutionContext executionContext = (DatabricksExecutionContext) context;
+        if (initFrom == null) {
+            executionContext.refreshDefaults(monitor, true);
+            return;
+        }
+
+        DatabricksExecutionContext executionMetaContext = (DatabricksExecutionContext) initFrom;
+        GenericCatalog defaultCatalog = executionMetaContext.getDefaultCatalog();
```

Comment:
Use getContextDefaults. No need to cast to DatabricksExecutionContext

Reply:
fixed